carp.client.core/src/commonMain/kotlin/dk/cachet/carp/client/domain/StudyRuntime.kt

```diff
@@ -108,24 +117,51 @@ class StudyRuntime private constructor(
      *
      * @return True in case deployment succeeded; false in case device could not yet be deployed (e.g., awaiting registration of other devices).
      * @throws UnsupportedOperationException in case deployment failed since not all necessary plugins to execute the study are available.
+     * @throws IllegalStateException in case data requested in the deployment cannot be collected on this client.
      */
-    suspend fun tryDeployment( deploymentService: DeploymentService ): Boolean
+    suspend fun tryDeployment( deploymentService: DeploymentService, dataCollector: DataCollector ): Boolean
     {
         val deploymentStatus = deploymentService.getStudyDeploymentStatus( studyDeploymentId )
-        val clientDeviceStatus = deploymentStatus.getDeviceStatus( device )
 
-        return tryDeployment( deploymentService, clientDeviceStatus )
+        return tryDeployment( deploymentService, dataCollector, deploymentStatus )
     }
 
-    private suspend fun tryDeployment( deploymentService: DeploymentService, deviceStatus: DeviceDeploymentStatus ): Boolean
+    private suspend fun tryDeployment(
+        deploymentService: DeploymentService,
+        dataCollector: DataCollector,
+        deploymentStatus: StudyDeploymentStatus
+    ): Boolean
     {
         // Early out in case state indicates the device is not yet ready to deploy.
+        val deviceStatus = deploymentStatus.getDeviceStatus( device )
         if ( deviceStatus !is DeviceDeploymentStatus.NotDeployed || !deviceStatus.isReadyForDeployment ) return false
 
-        // TODO: Get deployment and throw exception in case there are missing plugins to perform the operations (e.g., measurements).
+        // Get deployment information.
         val deployment = deploymentService.getDeviceDeploymentFor( studyDeploymentId, device.roleName )
+        check( deployment.deviceDescriptor == device )
         deploymentInformation = deployment
 
+        // Verify whether data can be collected for all requested measures.
+        for ( deviceTasks in deployment.getTasksPerDevice() )
+        {
+            val dataTypes = deviceTasks.tasks.flatMap { it.measures }.map { it.type }.distinct()
+            for ( type in dataTypes )
+            {
+                val canCollectData =
+                    if ( deviceTasks.isConnectedDevice )
```

Comment:
What does `isConnectedDevice` mean? Is it for external sensors?

Reply:
I'll add documentation to `DeviceTasks`. But yes, it is external sensors, corresponding to `StudyProtocol.addConnectedDevice`.

Reply:
Fixed in https://github.com/cph-cachet/carp.core-kotlin/pull/165/commits/b84075b5841c10a502961e49857aa52a2bbfa458